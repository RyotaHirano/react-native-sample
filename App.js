import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  Button,
  ScrollView
} from 'react-native';
import {
  StackNavigator
} from 'react-navigation'

// import PageA from './components/pageA.js'
// import PageB from './components/pageB.js'

// const PageAScreen = ({navigation}) => (
//   <PageA
//     navigation={navigation}
//   />
// )
//
// const PageBScreen = ({navigation}) => (
//   <PageB
//     navigation={navigation}
//   />
// )

const App = StackNavigator({
  PageA: {
    screen: PageAScreen
  },
  PageB: {
    screen: PageBScreen
  }
})

export default App