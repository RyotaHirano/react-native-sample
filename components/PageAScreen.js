import React from 'react'
import {
  Container, Content, Button, Text, Footer, FooterTab, Icon
} from 'native-base'

export default class PageAScreen extends React.Component {
  static navigationOptions = {
    title: 'PageA'
  }

  async componentWillMount() {
    await Expo.Font.loadAsync({
      'Roboto': require('native-base/Fonts/Roboto.ttf'),
      'Roboto_medium': require('native-base/Fonts/Roboto_medium.ttf'),
    });
  }

  render() {
    const { navigate } = this.props.navigation

    return (
      <Container>
        <Content>
          <Button
            primary
            title="PageB"
            onPress={ () => navigate('PageB') }
          >
            <Text>PageB</Text>
          </Button>
        </Content>
        <Footer>
          <FooterTab>
            <Button>
              <Icon ios='ios-add-circle' android='md-add-circle'/>
            </Button>
          </FooterTab>
        </Footer>
      </Container>
    )
  }
}