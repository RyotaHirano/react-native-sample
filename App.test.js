import 'react-native';
import { StackNavigator } from 'react-navigation'
import React from 'react';
import App from './App';
import PageAScreen from './components/PageAScreen'

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
});

test('render PageA', () => {
  const App = StackNavigator({
    PageA: {
      screen: PageAScreen
    },
  })
  const navigation = App

  let tree = renderer.create(
    <PageAScreen navigation={navigation}/>
  );

  tree = tree.toJSON()
  expect(tree).toMatchSnapshot();
})