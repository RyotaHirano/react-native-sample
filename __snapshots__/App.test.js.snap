// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PageA 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#007aff",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "lineHeight": 19,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          PageB
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderColor": "#cbcbcb",
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "center",
        "left": 0,
        "right": 0,
      }
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": undefined,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
      virtual={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 0,
            "flex": 1,
            "flexDirection": null,
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          android="md-add-circle"
          disabled={false}
          ellipsizeMode="tail"
          ios="ios-add-circle"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#6b6b6b",
                  "fontSize": 24,
                  "marginLeft": 16,
                  "marginRight": 16,
                  "paddingTop": 2,
                },
              ],
              Object {
                "fontFamily": "ExponentFont-mock-ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          virtual={undefined}
        >
          ÔÑÅ
        </Text>
      </View>
    </View>
  </View>
</View>
`;
